File: scala/src/main/scala/com/spoonacular/client/model/RecipesFindByIngredientsMissedIngredients.scala
package com.spoonacular.client.model

import io.circe._
import io.finch.circe._
import io.circe.generic.semiauto._
import io.circe.java8.time._
import spoonacular._
import java.math.BigDecimal
import scala.collection.immutable.Seq

/**
 * 
 * @param aisle 
 * @param amount 
 * @param id 
 * @param image 
 * @param meta 
 * @param name 
 * @param original 
 * @param originalName 
 * @param unit 
 * @param unitLong 
 * @param unitShort 
 */
case class RecipesFindByIngredientsMissedIngredients(aisle: String,
                amount: BigDecimal,
                id: Int,
                image: String,
                meta: Option[Seq[String]],
                name: String,
                original: String,
                originalName: String,
                unit: String,
                unitLong: String,
                unitShort: String
                )

object RecipesFindByIngredientsMissedIngredients {
    /**
     * Creates the codec for converting RecipesFindByIngredientsMissedIngredients from and to JSON.
     */
    implicit val decoder: Decoder[RecipesFindByIngredientsMissedIngredients] = deriveDecoder
    implicit val encoder: ObjectEncoder[RecipesFindByIngredientsMissedIngredients] = deriveEncoder
}

File: clojure/src/spoonacular_api/specs/_recipes_find_by_ingredients_missed_ingredients.clj
(ns spoonacular-api.specs.-recipes-find-by-ingredients-missed-ingredients
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            )
  (:import (java.io File)))


(def -recipes-find-by-ingredients-missed-ingredients-data
  {
   (ds/req :aisle) string?
   (ds/req :amount) float?
   (ds/req :id) int?
   (ds/req :image) string?
   (ds/opt :meta) (s/coll-of string?)
   (ds/req :name) string?
   (ds/req :original) string?
   (ds/req :originalName) string?
   (ds/req :unit) string?
   (ds/req :unitLong) string?
   (ds/req :unitShort) string?
   })

(def -recipes-find-by-ingredients-missed-ingredients-spec
  (ds/spec
    {:name ::-recipes-find-by-ingredients-missed-ingredients
     :spec -recipes-find-by-ingredients-missed-ingredients-data}))
